vaadin-combo-box-flow/src/main/java/com/vaadin/flow/component/combobox/ComboBox.java

```diff
@@ -868,13 +827,26 @@ public void setDataProvider(DataProvider<T, String> dataProvider) {
             userProvidedFilter = UserProvidedFilter.YES;
         }
 
+        // Fetch from data provider is enabled eagerly if the data provider
+        // is of in-memory type and it's not empty (no need to fetch from
+        // empty data provider). Otherwise, the fetch will be postponed until
+        // dropdown open event
+        final boolean enableFetch = InMemoryDataProvider.class
```

Comment:
What about `DataProvider::isInMemory` ??? If one has a custom in-memory data provider that is not based on `InMemoryDataProvider` ?

Reply:
Good catch! Thanks! Will fix

Reply:
Done